sdk/test-utils/recorder-new/MIGRATION.md

```diff
@@ -0,0 +1,160 @@
+# Migration Guide
+
+This document outlines key differences between the legacy recorder and the new Unified Recorder client.
+
+## Installing the Unified Recorder
+
+The first step is to install the unified recorder client library in your package by running
+
+```bash
+$ rush add --dev -p @azure-tools/test-recorder-new
+```
+
+You will use this library to control the recorder from your tests. The API is somewhat similar to the legacy recorder, albeit with some important differences. These will be discussed below.
+
+**⚠️ Important:** Do not uninstall the old recorder (`@azure-tools/test-recorder`) just yet! At this time, there are still utilities from the legacy recorder package that have not been migrated to the new package.
+
+## Changes to NPM scripts
+
+For the unified recorder client library to work, the [test proxy server] must be active while you are running your tests. Helpers have been added to the `dev-tool` package which manage starting and stopping the test proxy server before and after your tests are run. These helpers use the test proxy server Docker image. Therefore, Docker must be installed in your development environment.
+
+The script defined in `package.json` to run the browser tests should be replaced with `dev-tool run test:browser`, and the script to run the Node tests should look something like `dev tool run test:node-ts-input --mocha=\"--timeout 1200000 'test/**/*.spec.ts'`. In the case of the Node tests, the `--mocha="..."` parameter can be used to specify arguments to be passed through to Mocha. For example:
+
+| Script              | Before migration                                                                                                                                                                         | After migration                                                                                                                        |
+| :------------------ | :--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- | :------------------------------------------------------------------------------------------------------------------------------------- |
+| `unit-test:browser` | `karma start --single-run`                                                                                                                                                               | `dev-tool run test:browser`                                                                                                            |
+| `unit-test:node`    | `mocha -r esm -r ts-node/register --reporter ../../../common/tools/mocha-multi-reporter.js --timeout 1200000 --full-trace --exclude \"test/**/browser/*.spec.ts\" \"test/**/*.spec.ts\"` | `dev-tool run test:node-ts-input --mocha=\"--timeout 1200000 --full-trace --exclude 'test/**/browser/*.spec.ts' 'test/**/*.spec.ts'\"` |
```

Comment:
Add all the commands.
To help with a quick comparison.

Reply:
`integration-test` and `test`

Reply:
Didn't want to get too specific -- I feel like these will differ quite a bit between packages, no?

Reply:
Intention is to show how they are different amongst themselves.
`integration-test:node` vs `unit-test:node` and `test:node`.

I would prefer examples that are not opinionated, but that might be harder for understanding.

By adding all the changes here, at least the users would understand what to do for their commands from seeing the changes here.

Reply:
At least in the template package, those commands just call `unit-test` after a build. Changes shouldn't be required.

Reply:
And unrelated, but we'll have to address the code coverage issue (using `nyc`) at some point, since that's not working at the moment. It might require an addition to dev-tool?

Reply:
> And unrelated, but we'll have to address the code coverage issue (using nyc) at some point, since that's not working at the moment. It might require an addition to dev-tool?

Right, I was thinking simply support `integration-test:node-ts-input` and `integration-test:node-js-input` and etc.
But how about just `test:node-ts-input-nyc` and `test:node-ts-input`, similarly for js-input.

Reply:
I don't think I follow. Feel free to suggest the change you have in mind.